js2-mode.el

```diff
@@ -11373,6 +11373,17 @@ Selecting an error will jump it to the corresponding source-buffer error.
 
   (js2-reparse))
 
+;; We may eventually want js2-jsx-mode to derive from js-jsx-mode, but that'd be
+;; a bit more complicated and it doesn't net us much yet.
+;;;###autoload
+(define-derived-mode js2-jsx-mode js2-mode "JSX-IDE"
+  "Major mode for editing JSX code.
+Like `js-jsx-mode', which see.
+Indentation support requires Emacs 25+ (or the latest js.el)."
+  ;; Feature-detect JSX indentation support (in js-mode from Emacs 25+).
+  (if (fboundp 'js-jsx-indent-line)
```

Comment:
> Or copy over the new indentation code to js2-old-indent.el?

That's an option.

Or we just ask people to use `js2-minor-mode` with `js-jsx-mode`.

Or we put this mode in a separate file, and signal an error when it's loaded when the version of Emacs isn't high enough. Or simply signal this error from inside of `js2-jsx-mode`, and keep it in this file.

I think an error is better than a warning message.